crmsh/bootstrap.py

```diff
@@ -2335,10 +2436,18 @@ def bootstrap_join(context):
             _context.cluster_node = cluster_node
 
         join_ssh(cluster_node)
-        join_remote_auth(cluster_node)
-        join_csync2(cluster_node)
-        join_ssh_merge(cluster_node)
-        join_cluster(cluster_node)
+
+        lock_inst = JoinLock(cluster_node)
```

Comment:
We could put this in some kind of `context manager` or `decorator`

Reply:
Changed!